spokestack/asr/google/speech_recognizer.py

```diff
@@ -0,0 +1,128 @@
+"""
+This module contains the google asr speech recognizer
+"""
+import logging
+from queue import Queue
+from threading import Thread
+from typing import Any, Union
+
+import numpy as np  # type: ignore
+from google.cloud import speech  # type: ignore
+from google.oauth2 import service_account  # type: ignore
+
+from spokestack.context import SpeechContext
+
+
+_LOG = logging.getLogger(__name__)
+
+
+class GoogleSpeechRecognizer:
+    """Transforms speech into text using Google's ASR.
+
+    Args:
+        language (str): The language of given audio as a
+                        [BCP-47](https://www.rfc-editor.org/rfc/bcp/bcp47.txt)
+                        language tag. Example: "en-US"
+        credentials (Union[None, str, dict]): Dictionary of Google API credentials
+                                              or path to credentials. if set to None
+                                              credentials will be pulled from the
+                                              environment variable:
+                                              GOOGLE_APPLICATION_CREDENTIALS
+        sample_rate (int): sample rate of the input audio (Hz)
+        **kwargs (optional): additional keyword arguments
+    """
+
+    def __init__(
+        self,
+        language: str,
+        credentials=Union[None, str, dict],
```

Comment:
is this syntax correct?
seems like it should be `credentials: Union...`

Reply:
and it should probably default to `None`

Reply:
I fixed it.